modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -1747,20 +1747,24 @@ public final ImageWorker forceComponentColorModel() {
      * Reformats the {@linkplain ColorModel color model} to a {@linkplain ComponentColorModel
      * component color model} preserving transparency. This is used especially in order to go from
      * {@link PackedColorModel} to {@link ComponentColorModel}, which seems to be well accepted from
-     * {@code PNGEncoder} and {@code TIFFEncoder}.
+     * {@code PNGEncoder} and {@code TIFFEncoder}. The omitAlpha param allows to explicity avoid
+     * producing an alpha band when expanding an {@link IndexColorModel} to {@link
+     * ComponentColorModel}. This may be useful when preparing an Image for JPEG encoding which do
+     * not support alpha band, avoiding the need of a BandSelect right after the color expansion.
      *
      * <p>This code is adapted from jai-interests mailing list archive.
      *
      * @param checkTransparent
      * @param optimizeGray
+     * @param omitAlpha
      * @return this {@link ImageWorker}.
      * @see FormatDescriptor
      */
     public final ImageWorker forceComponentColorModel(
-            boolean checkTransparent, boolean optimizeGray) {
+            boolean checkTransparent, boolean optimizeGray, boolean omitAlpha) {
         final ColorModel cm = image.getColorModel();
         if (cm instanceof ComponentColorModel) {
-            // Already an component color model - nothing to do.
+            // Already a component color model - nothing to do.
```

Comment:
What if it was a component color model with alpha and omitAlpha was false? The javadoc could be made more explicit, and state that the color model is already component, then the alpha is preserved. Also, "omitAlpha" could be renamed to "omitAlphaOnExpand"

Reply:
Thx. Going to update it, accordingly.

Reply:
Done